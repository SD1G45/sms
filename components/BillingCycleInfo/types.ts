export interface BillingCycleInfoProps extends React.HTMLAttributes<HTMLInputElement> {
  title: string;
  viewLinkText: string;
  amount: string;
  subtitle: string;
}