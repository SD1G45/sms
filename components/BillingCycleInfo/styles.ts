import React from "react";
import styled from "styled-components";

export const SmallText = styled.text`
  text-align: left;
  width: 100%;
  color: #888B8E;
  font-size: 16px;
`;

export const LargeText = styled.text`
  text-align: left;
  width: 100%;
  font-size: 30px;
`;

export const LinkText = styled.text`
  text-align: right;
  width: 100%;
  color: #4881F0;
  font-size: 14px;
`;

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
  gap: 10%;
  margin-left: 10%;
  margin-right: 10%;
  padding: 0px;
`;

export const TitleDiv = styled.div`
  display: flex;
  width: 100%;
  padding: 0px;
  gap: 0%;
  margin: 0%;
`;

export const ColumnDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  padding: 30px 19px 30px 19px;
  gap: 11px;
`;