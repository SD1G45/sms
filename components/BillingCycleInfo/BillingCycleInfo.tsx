import React from "react";
import { ColumnDiv, LargeText, LinkText, SmallText, TitleDiv, Wrapper } from "./styles";
import { BillingCycleInfoProps } from "./types";

const BillingCycleInfo: React.FC<BillingCycleInfoProps> = ({
  title,
  viewLinkText,
  amount,
  subtitle
}) => {
  return (
    <Wrapper>
      <ColumnDiv>
        <TitleDiv>
          <SmallText>{title}</SmallText>
          <LinkText>{viewLinkText}</LinkText>
        </TitleDiv>
        <LargeText>{amount}</LargeText>
        <SmallText>{subtitle}</SmallText>
      </ColumnDiv>
    </Wrapper>
  );
};

export default BillingCycleInfo;