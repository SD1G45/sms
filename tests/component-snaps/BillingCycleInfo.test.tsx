import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import BillingCycleInfo from "../../components/BillingCycleInfo";

 const title = "title";
 const amount = "A lot";
 const subTitle= "sub";
 

 test("BillingCycleInfo renders correctly", () => {
     const tree = renderer.create(
      <BillingCycleInfo 
        title={title} 
        amount={amount} 
        subtitle={subTitle}
        linkRoute={"linkRoute"}
       viewLinkText={"text"}/>).toJSON();
      expect(tree).toMatchSnapshot();
 })

