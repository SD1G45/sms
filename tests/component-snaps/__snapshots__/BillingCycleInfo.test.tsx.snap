// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingCycleInfo renders correctly 1`] = `
.c3 {
  text-align: left;
  width: 100%;
  color: #888B8E;
  font-size: 16px;
}

.c5 {
  text-align: left;
  width: 100%;
  font-size: 30px;
}

.c4 {
  text-align: right;
  width: 100%;
  color: #4881F0;
  font-size: 14px;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  gap: 10%;
  margin-left: 10%;
  margin-right: 10%;
  padding: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding: 0px;
  gap: 0%;
  margin: 0%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px 19px 30px 19px;
  gap: 11px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <text
        className="c3"
      >
        title
      </text>
      <text
        className="c4"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        text
      </text>
    </div>
    <text
      className="c5"
    >
      A lot
    </text>
    <text
      className="c3"
    >
      sub
    </text>
  </div>
</div>
`;
